tokenExchange/src/app/tokenGen.js

```diff
@@ -1,27 +1,16 @@
 const AWS = require("aws-sdk");
 const kms = new AWS.KMS();
 const base64url = require("base64url");
-const properties = require('./properties.js')
 
-async function sign(tokenParts, keyId) {
-    let message = Buffer.from(tokenParts.header + "." + tokenParts.payload)
-
-    let res = await kms.sign({
-        Message: message,
-        KeyId: keyId,
-        SigningAlgorithm: 'RSASSA_PKCS1_V1_5_SHA_256',
-        MessageType: 'RAW' 
-    }).promise()
-
-    tokenParts.signature = res.Signature.toString("base64")
-    .replace(/\+/g, '-')
-    .replace(/\//g, '_')
-    .replace(/=/g, '');
-    
-    let token = tokenParts.header + "." + tokenParts.payload + "." + tokenParts.signature;
-    console.debug('token ', token);
 
-    return token;
+module.exports = {
+    async generateToken(decodedToken){
+        const keyId = process.env.KEY_ID;
+        let token_components = getTokenComponent(decodedToken);
+        let res = await sign(token_components, keyId)
```

Comment:
Una riga di log ta la riga 9 e la 10 potrebbe essere utile.
In generale vedo poc loggato a info